app/modules/adc.c

```diff
@@ -24,10 +24,49 @@ static int adc_readvdd33( lua_State* L )
   return 1;
 }
 
+// Lua: adc.force_init_byte107(x)
+static int adc_init107( lua_State *L )
+{
+  uint8_t byte107 = luaL_checkinteger (L, 1);
+
+  uint32 init_sector = flash_safe_get_sec_num () - 4;
+
+  // Note 32bit alignment so we can safely cast to uint32 for the flash api
+  char init_data[SPI_FLASH_SEC_SIZE] __attribute__((aligned(4)));
+
+  if (SPI_FLASH_RESULT_OK != flash_safe_read (
+    init_sector * SPI_FLASH_SEC_SIZE,
+    (uint32 *)init_data, sizeof(init_data)))
+      return luaL_error(L, "flash read error");
+
+  // If it's already the correct value, we don't need to force it
+  if (init_data[107] == byte107)
+  {
+    lua_pushboolean (L, false);
+    return 1;
+  }
+
+  // Nope, it differs, we need to rewrite it
+  init_data[107] = byte107;
+  if (SPI_FLASH_RESULT_OK != flash_safe_erase_sector (init_sector))
+    return luaL_error(L, "flash erase error");
+
+  if (SPI_FLASH_RESULT_OK != flash_safe_write (
+    init_sector * SPI_FLASH_SEC_SIZE,
+    (uint32 *)init_data, sizeof(init_data)))
+      return luaL_error(L, "flash write error");
+
+  lua_pushboolean (L, true);
+  return 1;
+}
+
 // Module function map
 static const LUA_REG_TYPE adc_map[] = {
   { LSTRKEY( "read" ),      LFUNCVAL( adc_sample ) },
-  { LSTRKEY( "readvdd33" ), LFUNCVAL( adc_readvdd33) },
+  { LSTRKEY( "readvdd33" ), LFUNCVAL( adc_readvdd33 ) },
+  { LSTRKEY( "force_init_byte107" ), LFUNCVAL( adc_init107 ) },
```

Comment:
I actually did think about that for a few seconds but I dismissed it because (in contrast to you) I didn't find a sensible and logical explanation for it 😉 Your reasoning makes a lot of sense, go for it.